### ISSUE TITLE
Authorization header not forwarded when using Eureka Service Discovery

### ISSUE THREAD

**Issue by michaelstievenart** [micronaut-projects/micronaut-core#3553]
### Steps to Reproduce

1. Running the Eureka Server
2. Running a Micronaut service (token propagation is setup)
3. Running A Spring Boot service/Micronaut service
4. Running an Auth Server that is generating bearer tokens.

### Expected Behaviour
When service A (Micronaut service) makes a request to service B (Spring boot or Micronaut Service) the Token TokenPropagationHttpClientFilter successfully adds the Authorization Header and forward the request to service B using eureka discovery. When request arrives at service B the Authorization header is present.

### Actual Behaviour
When service A (Micronaut service) makes a request to service B (Spring boot or Micronaut Service) the Token TokenPropagationHttpClientFilter successfully adds the Authorization Header and forward the request to service B using eureka discovery. When request arrives at service B the Authorization header is missing.

### Environment Information

- Operating System**: Windows
- Micronaut Version: 2.0.0.M3
- JDK Version: 12

### Example Application
Will post an example application here tomorrow, still putting it together

**Comment by michaelstievenart** [comment]
Put the repo together demonstrating issue.
https://github.com/michaelstievenart/micronaut-eureka-service-discovery
I initially found the issue on Micronaut Version: 2.0.0.M3 but I have generated the example repo on 2.0.0.RC2

**Comment by graemerocher** [comment]
Will investigate thanks

**Comment by jameskleeh** [comment]
@michaelstievenart This is working as designed as far as I can tell. You have `service-id-regex` set to `eureka`, however the service id of the target server is `appB`, so that doesn't match. Even if I set the `uri-regex` to `.*` so the token will be passed through, you have no signature configurations on appB to validate the JWT.

**Comment by michaelstievenart** [comment]
@jameskleeh Hi I just want to clarify something with you around this.

1. Apologies you are correct, I forgot to annotate the RSASignatureConfiguration with the service id in APP B.
I have corrected this, and tested service B individually and it is able to authenticate as expected.
I also set the service id regex on service A to .*

1. I set the service id regex to eureka as eureka is the service id returned from the OutgoingHttpRequestProcessorImpl shouldProcessRequest when I attempt to make a request from service A to service B, so far this is the case when I have attempted to make any service request using eureka.
2. As mentioned I also have corrected for the missing RSASignatureConfiguration in app B.
From all the internal logging as well as look at the CustomFilter (OncePerRequestHttpServerFilter) which I gave highest precedence, there is no Bearer Token on the request when it arrives at service B.

I am not following how this is expected behavior.
